### PR TITLE
[WIP] Add proper settings handling

### PR DESCRIPTION
**Pull request by rixx**
- [x] Use proper settings structure
- [ ] Add tests?
- [x] Add example settings file
- [x] Add documentation
- [x] Add compatibility for older structure

### CODE REVIEW

File: src/pretalx/settings.py [hunk 1]
```diff
@@ -11,28 +10,98 @@
...
-encoding='utf-8')
+    config.read([
+        '/etc/pretalx/pretalx.cfg',
+        os.path.expanduser('~/.pretalx.cfg'),
+        'pretalx.cfg',
+    ], encoding='utf-8')
+
+env_config = {
+    'filesystem': {
+        'data': os.getenv('PRETALX_DATA_DIR'),
```
Comment (luto) [inline src/pretalx/settings.py]: Most of these env variables are introduced in this MR, right? Is there a reason to not make them match the config file options? e.g. `[filesystem] data` => `PRETALX_FILESYSTEM_DATA`.

File: src/pretalx/settings.py [hunk 2]
```diff
@@ -11,28 +10,98 @@
 
 
 config = configparser.RawConfigParser()
+config.read_dict({
+    'filesystem': {
+        'base': os.path.dirname(os.path.dirname(__file__)),
+    },  # defaults depend on the data dir and need to be set once the data dir is fixed
+    'site': {
+        'debug': 'runserver' in sys.argv,
+        'url': 'http://localhost',
```
Comment (luto) [inline src/pretalx/settings.py]: does this somehow take the port into account later on?

Comment (rixx) [reply src/pretalx/settings.py]: Nope, but I'd like to tackle that problem later if at all

File: src/pretalx/settings.py (hunk 1 above)
Comment (rixx) [reply src/pretalx/settings.py]: Nope, most of these options have been around already, which is the reason for the somewhat inconsistent naming.

**Pull request merged**